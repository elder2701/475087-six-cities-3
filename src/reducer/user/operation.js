import {ActionCreator, AuthorizationStatus} from "./user.js";

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
        .then((res) => {
          dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
          dispatch(ActionCreator.loadUserInfo(res.data));
        })
        .catch((err) => {
          throw err;
        });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
    .then((res) => {
      dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      dispatch(ActionCreator.loadUserInfo(res.data));
    });
  },
};

export {Operation};
