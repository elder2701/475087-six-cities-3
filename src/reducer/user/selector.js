import NameSpace from "../name-space.js";

const getAuthStatus = (state)=> state[NameSpace.USER].authorizationStatus;

export {getAuthStatus};
