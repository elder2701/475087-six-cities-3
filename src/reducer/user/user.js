import {extend} from "../../utils.js";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.AUTH,
  userInfo: {}
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  LOAD_USER_INFO: `LOAD_USER_INFO`
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: status,
    };
  },
  loadUserInfo: (userInfo) => {
    return {
      type: ActionType.LOAD_USER_INFO,
      payload: userInfo
    };
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });

    case ActionType.LOAD_USER_INFO:
      return extend(state, {
        userInfo: action.payload
      });
  }
  return state;
};

export {ActionCreator, reducer, AuthorizationStatus};
