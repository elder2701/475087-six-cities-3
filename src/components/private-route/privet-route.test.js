import React from "react";
import renderer from "react-test-renderer";
import {PrivateRoute} from "./private-route.jsx";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {Router} from "react-router-dom";
import history from "../../history.js";

it(`<PlaceReviews /> sould be render auth`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <PrivateRoute
            render={() => <div></div>}
            authStatus={AuthorizationStatus.AUTH}
            path={`/`}
            exact
          />
        </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`<PlaceReviews /> sould be render not auth`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <PrivateRoute
            render={() => <div></div>}
            authStatus={AuthorizationStatus.NO_AUTH}
            path={`/`}
            exact
          />
        </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
