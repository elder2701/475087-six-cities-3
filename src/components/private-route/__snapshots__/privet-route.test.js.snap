// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceReviews /> sould be render auth 1`] = `<div />`;

exports[`<PlaceReviews /> sould be render not auth 1`] = `null`;
